modules/search-service-impl/src/main/java/org/opencastproject/search/impl/SearchServiceIndex.java

```diff
@@ -0,0 +1,481 @@
+      throw new SearchException(e);
+    }
+  }
+
+  /**
+   * Immediately removes the given series from the search service.
+   *
+   * @param seriesId
+   *          the series
+   * @throws SearchException
+   */
+  public boolean deleteSeriesSynchronously(String seriesId) throws SearchException {
+    try {
+      logger.info("Marking {} as deleted in the search index", seriesId);
+      JsonElement json = gson.toJsonTree(Map.of(
+          "deleted", Instant.now().getEpochSecond(),
+          "modified", Instant.now().toString()));
+      var updateRequest = new UpdateRequest(INDEX_NAME, seriesId).doc(gson.toJson(json), XContentType.JSON);
+      UpdateResponse response = esIndex.getClient().update(updateRequest, RequestOptions.DEFAULT);
+      //NB: We're marking things as deleted but *not actually deleting them**
+      return DocWriteResponse.Result.UPDATED == response.getResult();
+    } catch (IOException e) {
+      throw new SearchException("Could not delete series " + seriesId + " from index", e);
+    }
+  }
+
+  @Override
+  public void repopulate() throws IndexRebuildException {
+    try {
+      int total = persistence.countMediaPackages();
+      int pageSize = 50;
+      int offset = 0;
+      AtomicInteger current = new AtomicInteger(0);
+      logIndexRebuildBegin(logger, esIndex.getIndexName(), total, "search");
+      List<Tuple<MediaPackage, String>> page = null;
+
+      do {
+        page = persistence.getAllMediaPackages(pageSize, offset).collect(Collectors.toList());
+        page.forEach(tuple -> {
+          try {
+            MediaPackage mediaPackage = tuple.getA();
+            String mediaPackageId = mediaPackage.getIdentifier().toString();
+
+            AccessControlList acl = persistence.getAccessControlList(mediaPackageId);
+            Date modificationDate = persistence.getModificationDate(mediaPackageId);
+            Date deletionDate = persistence.getDeletionDate(mediaPackageId);
+
+            AccessControlList seriesAcl = persistence.getAccessControlLists(mediaPackage.getSeries(), mediaPackageId)
+                .stream().reduce(new AccessControlList(acl.getEntries()), AccessControlList::mergeActions);
+            logger.debug("Updating series ACL with merged access control list: {}", seriesAcl);
+
+            logIndexRebuildProgress(logger, esIndex.getIndexName(), total, current.getAndIncrement(), pageSize);
+            indexMediaPackage(mediaPackage, acl, modificationDate, deletionDate);
+          } catch (SearchServiceDatabaseException | UnauthorizedException | NotFoundException e) {
+            logIndexRebuildError(logger, "search", total, current.get(), e);
+            //NB: Runtime exception thrown to escape the functional interfacing
+            throw new RuntimeException("Internal Index Rebuild Failure", e);
+          }
+        });
+        offset += pageSize;
+      } while (page.size() == pageSize);
```

Comment:
This should probably be:
```suggestion
      } while (offset <= total);
```